schema_utils.py

```diff
@@ -104,10 +108,40 @@ def normalize_against_schema(
         assert isinstance(obj, dict), ('Expected dict, received %s' % obj)
         expected_dict_keys = [
             p[SCHEMA_KEY_NAME] for p in schema[SCHEMA_KEY_PROPERTIES]]
+
+        # For handling cases of missing args.
+        list_of_missing_args_with_default_none = []
+        for prop in schema[SCHEMA_KEY_PROPERTIES]:
+
+            if (SCHEMA_KEY_DEFAULT_VALUE not in prop[SCHEMA_KEY_SCHEMA] or
+                    prop[SCHEMA_KEY_NAME] in obj):
+                continue
+
+            default_value = prop[SCHEMA_KEY_SCHEMA][SCHEMA_KEY_DEFAULT_VALUE]
+
+            # Don't normalize if default_value is None.
+            if default_value is None:
+                expected_dict_keys.remove(prop[SCHEMA_KEY_NAME])
+                list_of_missing_args_with_default_none.append(
+                    prop[SCHEMA_KEY_NAME])
+                continue
+            obj[prop[SCHEMA_KEY_NAME]] = default_value
+
+        # To remove dict from the properties field if the schema of a
+        # missing arg contains 'default_value': None.
+        for _ in list_of_missing_args_with_default_none:
+            for i in python_utils.RANGE(len(schema[SCHEMA_KEY_PROPERTIES])):
+                if (schema[SCHEMA_KEY_PROPERTIES][i][SCHEMA_KEY_NAME] not in
+                        list_of_missing_args_with_default_none):
+                    continue
+                del schema[SCHEMA_KEY_PROPERTIES][i]
+                break
```

Comment:
Why are we doing this? I don't think we should change the structure of the dict just because the default_value of the field is None. Are we trying to handle the optional arg case through default_value: None?

Reply:
Yes, optional cases are also handled with ‘’default_value’’ key, when the default_value key is set to None value.

Reply:
Any specific reason for doing it this way? I mean is there any issue with having "optional" key? "Treating `"default: None` to indicate optional field doesn't seem intuitive to me!

Reply:
If you have an optional key, then there's a potential conflict where optional is True and default is not None. So, keeping it in one key seems simpler (if the optional value isn't provided then it should be reasonable for us to regard it as None for the purposes of subsequent computations).

Reply:
Thank you @seanlip.
Adding a note on this, we also have a wiki page which clearly mentions the conventions that we have adopted regarding schema validation.
Link: https://github.com/oppia/oppia/wiki/Validation-of-handler-args#schema-keys
@DubeySandeep please take a look, thanking you.